File: backend/src/main/java/com/kineticrehab/service/PacienteService.java
package com.kineticrehab.service;

import com.kineticrehab.dto.request.PacienteRequestDTO;
import com.kineticrehab.dto.response.PacienteResponseDTO;

import java.util.List;

public interface PacienteService {

    List<PacienteResponseDTO> listarTodos();

    PacienteResponseDTO buscarPorId(Long id);

    PacienteResponseDTO crear(PacienteRequestDTO dto);

    PacienteResponseDTO actualizar(Long id, PacienteRequestDTO dto);

    void eliminar(Long id);

    List<PacienteResponseDTO> buscarPorTermino(String termino);
}

File: frontend/src/app/core/services/paciente.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Paciente, PacienteRequest } from '../../models/paciente.model';

@Injectable({ providedIn: 'root' })
export class PacienteService {

  private readonly API_URL = '/api/pacientes';

  constructor(private http: HttpClient) {}

  listarTodos(): Observable<Paciente[]> {
    return this.http.get<Paciente[]>(this.API_URL);
  }

  buscarPorId(id: number): Observable<Paciente> {
    return this.http.get<Paciente>(`${this.API_URL}/${id}`);
  }

  buscar(termino: string): Observable<Paciente[]> {
    return this.http.get<Paciente[]>(`${this.API_URL}/buscar`, { params: new HttpParams().set('q', termino) });
  }

  crear(dto: PacienteRequest): Observable<Paciente> {
    return this.http.post<Paciente>(this.API_URL, dto);
  }

  actualizar(id: number, dto: PacienteRequest): Observable<Paciente> {
    return this.http.put<Paciente>(`${this.API_URL}/${id}`, dto);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.API_URL}/${id}`);
  }
}
